### PR TITLE
[IoTDB-468] Restructure QueryPlan

### PR DESCRIPTION
**Pull request by Alima777**
The AlignByDevice query ( which is called groupByDevice before ) and general query aligning by time are both storaged in QueryPlan now. However, they invoke different query process bascially, so it's necessary to restructure the QueryPlan and seperate the AlignByDeviceQuery from it. 

The QueryPlan is designed as base class, which has AlignByTimePlan and AlignByDevicePlan as subclasses. This restructure may bring some usage changes, because the general query which uses QueryPlan directly before has to instantiate AlignByTimePlan now.

If you think the design is not friendly for users, welcome to comment.

### CODE REVIEW

File: server/src/main/java/org/apache/iotdb/db/qp/physical/crud/AlignByDevicePlan.java [hunk 1]
```diff
@@ -0,0 +1,162 @@
+package org.apache.iotdb.db.qp.physical.crud;
```
Comment (samperson1997) [inline server/src/main/java/org/apache/iotdb/db/qp/physical/crud/AlignByDevicePlan.java]: Hi, don't forget the Apache license.

File: server/src/main/java/org/apache/iotdb/db/qp/physical/crud/AlignByTimePlan.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
+package org.apache.iotdb.db.qp.physical.crud;
```
Comment (samperson1997) [inline server/src/main/java/org/apache/iotdb/db/qp/physical/crud/AlignByTimePlan.java]: Same above.

**Comment by qiaojialin** [comment]
resubmit to new_series_reader

File: server/src/main/java/org/apache/iotdb/db/qp/physical/crud/AlignByDevicePlan.java (hunk 1 above)
Comment (Alima777) [reply server/src/main/java/org/apache/iotdb/db/qp/physical/crud/AlignByDevicePlan.java]: Thank you! I will fix it.

File: server/src/main/java/org/apache/iotdb/db/qp/physical/crud/AlignByTimePlan.java (hunk 2 above)
Comment (Alima777) [reply server/src/main/java/org/apache/iotdb/db/qp/physical/crud/AlignByTimePlan.java]: Thank you! I will fix it.

**Pull request closed without merging**